test/chart/checkRequired.js

```diff
@@ -0,0 +1,83 @@
+import jsdom from 'jsdom';
+import createChart from '../../src/createChart';
+import checkRequired from '../../src/chart/checkRequired';
+import expect from 'expect';
+import settings from '../samples/irisSettings';
+import data from '../samples/irisData';
+import d3 from 'd3';
+
+describe('upon chart initialization data field arguments are checked against actual fields in data', () => {
+    const { JSDOM } = jsdom;
+    let dom,
+        container,
+        chart;
+
+  //DOM setup
+    before(() => {
+        dom = new JSDOM('<!DOCTYPE html>');
+        container = dom.window.document.createElement('div');
+    });
+
+  //Chart initialization
+    beforeEach(() => {
+        chart = createChart(container, settings)
+            .init(data, true);
+    });
+
```

Comment:
`per` is checked starting on line 41.  `tooltips` and `text` are not checked but at the same time they won't break a chart.